cms/tests/test_views.py

```diff
@@ -269,6 +272,32 @@ def test_incorrect_slug_for_language(self):
         self.assertRedirects(response, '/de/jobs/')
 
 
+    def test_page_sanitisation_xss_attack(self):
+        """
+            When sending a request the CMS uses get_page_from_request to return the appropriate page.
+            None should be returned
+        """
+        request = self.get_request("/")
+        request.path_info = "<script>alert('attack!')</script>"
+
+        response = get_page_from_request(request)
+
+        # If this method is passed a parameter which is not a valid primary key
+        # for a page object nothing should be returned.
+        self.assertEqual(response, None)
+
+    def test_malicious_content_login_request(self):
+        user = self.get_superuser()
+        request = self.get_request(
+            "/en/admin/login/?q=<script>alert('Attack')</script>",
+            post_data={"username": user.username, "password": user.username}
```

Comment:
```suggestion
            post_data={"username": username, "password": username}
```

@adam-murray This breaks the test suite, I'm not sure why the username field changes for just the Python 3.6 versions. To fix the test suite this needs to be:

```
username = getattr(user, get_user_model().USERNAME_FIELD)
```

Taken from: https://github.com/django-cms/django-cms/blob/f30f0204a8c3e498aa86de0aba35470521859873/cms/tests/test_no_i18n.py#L121

**With the suggestions above you'll still need to add the get_user_model() import at the top of the file:**
```
from django.contrib.auth import get_user_model
```